Lib/fontbakery/profiles/gdef.py

```diff
@@ -0,0 +1,124 @@
+from fontbakery.callable import check
+from fontbakery.checkrunner import FAIL, PASS, WARN
+from fontbakery.message import Message
+from fontbakery.utils import pretty_print_list
+
+# used to inform get_module_profile whether and how to create a profile
+from fontbakery.fonts_profile import profile_factory # NOQA pylint: disable=unused-import
+
+from fontTools import unicodedata
+
+
+GDEF_MARK_GLYPH_CLASS = 3
+
+
+@ check(
+  id = 'com.google.fonts/check/gdef_spacing_marks',
+  rationale = """
+    Glyphs in the GDEF mark glyph class should be non-spacing.
+    Spacing glyphs in the GDEF mark glyph class may have incorrect
+    anchor positioning that was only intended for building composite glyphs
+    during design.
+  """
+)
+def com_google_fonts_check_gdef_spacing_marks(ttFont):
+  """Check mark characters are in GDEF mark glyph class)"""
+  if "GDEF" in ttFont:
+    spacing_glyphs = {name for (name, value) in ttFont["hmtx"].metrics.items()
+                      if value}
+    class_defs = ttFont["GDEF"].table.GlyphClassDef.classDefs.items()
+    mark_class_glyphs = set([name for (name, value) in class_defs
+                             if value == GDEF_MARK_GLYPH_CLASS])
```

Comment:
the code-linter is being a bit too opinionated here, but I think that when it says "consider using a set comprehension", it means something similar to this:
```python
mark_class_glyphs = {name for (name, value) in class_defs
                     if value == GDEF_MARK_GLYPH_CLASS}
```

https://python-reference.readthedocs.io/en/latest/docs/comprehensions/set_comprehension.html